src/app/qgisapp.cpp

```diff
@@ -16698,21 +16694,37 @@ void QgisApp::projectChanged( const QDomDocument &doc )
 }
 
 #ifdef HAVE_3D
-void QgisApp::write3DMapViewSettings( Qgs3DMapCanvasDockWidget *w, QDomDocument &doc, QDomElement &elem3DMap )
+void QgisApp::write3DMapViewSettings( Qgs3DMapCanvasWidget *widget, QDomDocument &doc, QDomElement &elem3DMap )
 {
   QgsReadWriteContext readWriteContext;
   readWriteContext.setPathResolver( QgsProject::instance()->pathResolver() );
-  elem3DMap.setAttribute( QStringLiteral( "name" ), w->mapCanvas3D()->objectName() );
-  QDomElement elem3DMapSettings = w->mapCanvas3D()->map()->writeXml( doc, readWriteContext );
+  elem3DMap.setAttribute( QStringLiteral( "name" ), widget->canvasName() );
+  QDomElement elem3DMapSettings = widget->mapCanvas3D()->map()->writeXml( doc, readWriteContext );
   elem3DMap.appendChild( elem3DMapSettings );
-  QDomElement elemCamera = w->mapCanvas3D()->cameraController()->writeXml( doc );
+  QDomElement elemCamera = widget->mapCanvas3D()->cameraController()->writeXml( doc );
   elem3DMap.appendChild( elemCamera );
-  QDomElement elemAnimation = w->animationWidget()->animation().writeXml( doc );
+  QDomElement elemAnimation = widget->animationWidget()->animation().writeXml( doc );
   elem3DMap.appendChild( elemAnimation );
-  writeDockWidgetSettings( w, elem3DMap );
+
+  QgsDockableWidgetHelper *w = widget->dockableWidgetHelper();
+  elem3DMap.setAttribute( QStringLiteral( "isDocked" ), w->isDocked() );
+
+  QRect dockGeom = w->dockGeometry();
+  elem3DMap.setAttribute( QStringLiteral( "x" ), dockGeom.x() );
+  elem3DMap.setAttribute( QStringLiteral( "y" ), dockGeom.y() );
+  elem3DMap.setAttribute( QStringLiteral( "width" ), dockGeom.width() );
+  elem3DMap.setAttribute( QStringLiteral( "height" ), dockGeom.height() );
+  elem3DMap.setAttribute( QStringLiteral( "floating" ), w->isDockFloating() );
+  elem3DMap.setAttribute( QStringLiteral( "area" ), w->dockFloatingArea() );
+
+  QRect dialogGeom = w->dialogGeometry();
+  elem3DMap.setAttribute( QStringLiteral( "d_x" ), dialogGeom.x() );
+  elem3DMap.setAttribute( QStringLiteral( "d_y" ), dialogGeom.y() );
+  elem3DMap.setAttribute( QStringLiteral( "d_width" ), dialogGeom.width() );
+  elem3DMap.setAttribute( QStringLiteral( "d_height" ), dialogGeom.height() );
```

Comment:
all this code should be encapsulated in a call `widget->dockableWidgetHelper()->writeXml( elem3DMap );` - because other dockable widgets will need the same code